clustered/server/service/src/main/java/org/ehcache/clustered/server/state/EhcacheStateServiceProvider.java

```diff
@@ -66,14 +66,14 @@ public boolean initialize(ServiceProviderConfiguration configuration, PlatformCo
       throw new UnsupportedOperationException("There are " + extendedConfiguration.size() + " OffHeapResourcesProvider, this is not supported. " +
         "There must be only one!");
     }
-    Iterator<OffHeapResources> iterator = extendedConfiguration.iterator();
-    if (iterator.hasNext()) {
-      offHeapResourcesProvider = iterator.next();
-      if (offHeapResourcesProvider.getAllIdentifiers().isEmpty()) {
-        LOGGER.warn("No offheap-resource defined - this will prevent provider from offering any EhcacheStateService.");
-      }
-    } else {
-      throw new UnsupportedOperationException("There are no offheap-resource defined, this is not supported");
+    try {
+      // When a server is activated, there will ALWAYS be one OffHeapResources, that will hold the mapping configured by the user with the "offheap-resources" setting.
+      offHeapResourcesProvider = extendedConfiguration.iterator().next();
+    } catch (NoSuchElementException ignored) {
+      // In diagnostic mode, no extended configuration is loaded, so there won't be any OffHeapResources.
```

Comment:
Would we be better off to just keep the existing structure and change the `throw new UnsupportedOperationException(...)` for a logger call - then we also log something different when the "no resources" is due to the "no service".

Reply:
Frankly, what would really be better is to `return false` instead to leverage platform's service loading capabilities and have platform discard the service. In diagnostic mode anyway, nothing is there to call `getService()`. This is what @myronkscott was also suggesting... I would prefer to change to that if possible ?

Reply:
seems like the correct behavior but does this cascade into failure somewhere else in bootstrapping just in case this is post bootstrap and no off heap is defined?

Reply:
No because nothing is calling `getService()` in diagnostic mode. And when activated, if no offheap, entity creation service is already having `throw new ConfigurationException("Unable to retrieve EhcacheStateService: " + e.getMessage());` in case no service is found.